### DISCUSSION TITLE
Send a position struct

### DISCUSSION THREAD

**Discussion by kockie69** [mpaperno/WASimCommander#14] [Q&A]
Hi,

I have a position struct that looks like:

struct Position
{
	bool onGround;
	double pitch;
	double bank;
	double  altitude;
	double  longitude;
	double  latitude;
	double  true_heading;
};

I would like to pass the values of this position in one call to FS2020. How can I do this in WaSim? I can make individual calls but that isn't the perfect way (I think). WaSim taking care of this and packages these multiple calls into one?

**Comment by mpaperno** [comment]
Hi!

I guess my first question is what would you like to do with this data on the sim side? Once WASimModule receives the struct data...  assuming we can get the data there one way or another.

There are currently no functions/features to replicate the SimConnect struct-type variable handling (like those listed in https://docs.flightsimulator.com/html/Programming_Tools/SimVars/Simulation_Variable_Units.htm#structs_and_oth), in particular because there's no way, using the WASM "Gauge API" functions, to read or write multiple values/variables at once. As far as I can tell, those SimConnect structs are specific to something SimConnect does, which ultimately is probably reading all those values individually and then just sending them back as one struct vs. individually.

Your struct looks like `latlonaltpbh` with extra fields...?  And are you trying to set these values, or just read them back all together in an efficient manner?  I think requesting them individually will not be any more resource intensive than packing them into a struct, and may actually be more efficient in some cases (only benchmarks would really tell us though).

Anyway, let me know if any of that hits the mark.

Cheers,
-Max
